### ISSUE TITLE
Gap between UINavigationBar and ViewController on iPhone 14 Pro Max

### ISSUE THREAD

**Issue by fredrikp** [LeoNatan/LNPopupController#518]
This issue occurs only on the iPhone 14 Pro Max. After calling presentPopupBar, a gap appears between the navigation bar and the contents of the displayed view controller in the app I'm working on. After calling dismissPopupBar, the view controller moves up and the gap disappears.

The popupbar is presented on a UITabBarController. Attached is a screenshot from the app I'm working on running on the iOS simulator. Both are running iOS 16.0 and LNPopupController version 2.13.7.

I tried to run the example project contained in LNPopupController to see if the gap appeared there as well. It did not, but another effect was visible. If you run the example project on an iPhone 14 Pro Max, you can see the contents "jumping" immediately after startup when the popup is displayed. I have attached a small clip showing this happening as well. Running the example project on an iPhone 13 Pro Max is smooth with no jumping.

![Example](https://user-images.githubusercontent.com/888442/192480697-31123c8b-58dc-4c61-bf99-614b1c2beef8.jpg)

https://user-images.githubusercontent.com/888442/192484828-2abac0d0-d32f-4067-9abf-57d8ec44d8d7.mp4

**Comment by LeoNatan** [comment]
I will take a look soon. Thanks

**Comment by LeoNatan** [comment]
The “jump” in the video is unrelated to my framework, but seems like Apple’s launch screen renderer (from the launch storyboard in the app) is buggy.

I cannot reproduce the main issue. From my testing, safe area insets are correctly set:
```
Safe area insets: UIEdgeInsets: {59, 0, 34, 0}
```

**Comment by LeoNatan** [comment]
Please provide more information with regards to how you set up your hierarchy. It might be a UIKit issue. But since I cannot reproduce, I can’t know for sure.

**Comment by LeoNatan** [comment]
You haven’t been active. Please try with my changes in the latest build, and see if that fixes things. If not, please follow requests above.

**Comment by fredrikp** [comment]
I updated from 2.13.7 to 2.14.0 but unfortunately I see the same issue in my app. I will attempt to make a standalone demo project to illustrate the issue, but this may take some time.

In the mean time, the hierarchy in my app is as follows:

```
UIWindow
- UIViewController
- - UITabBarController, presented with .currentContext style
- - - UINavigationController for each tab
- - - - UIViewController inside navcontroller for each tab 
- - - Custom popupBar
```

**Comment by LeoNatan** [comment]
If you are able to reproduce, please attach it here and we'll reopen the issue. Thanks

**Comment by fredrikp** [comment]
While creating a project to reproduce the issue I found out what caused it. 

When my app displays the popup bar it is always together with a presentation of another UIViewController that contains a chat application. When this view controller is dismissed, the popup bar becomes visible, displaying chat status and unread message. 

I had added a delay of 0.1 seconds before displaying the popup bar in order to avoid it being visible for a short time before being covered again by the presented chat view controller.

I guess that this means that your library's size calculations were performed while a view transition was in effect, which led to sizes that were slightly off. Simply removing the delay in my code made the black bar disappear. For some reason, this issue never occurred before the iPhone 14 Pro Max was released.

So no need to reopen the issue. Thanks for your assistance!
